MSAL/src/requests/MSALBaseRequest.m

```diff
@@ -164,9 +165,79 @@ - (void)resolveEndpoints:(MSALAuthorityCompletion)completionBlock
 
 - (void)acquireToken:(nonnull MSALCompletionBlock)completionBlock
 {
-    NSMutableDictionary<NSString *, NSString *> *reqParameters = [NSMutableDictionary new];
+    MSIDTokenRequest *authRequest = [self tokenRequest];
     
-    // TODO: Remove once uid+utid work hits PROD
+    [authRequest sendWithBlock:^(id response, NSError *error) {
+        
+        [authRequest finishAndInvalidate];
+        
+        if (error)
+        {
+            completionBlock(nil, error);
+            return;
+        }
+        
+        if(response && ![response isKindOfClass:[NSMutableDictionary class]])
+        {
+            NSError *localError = CREATE_MSID_LOG_ERROR(_parameters, MSALErrorInternal, @"response is not of the expected type: NSMutableDictionary.");
+            completionBlock(nil, localError);
+            return;
+        }
+        
+        NSMutableDictionary *jsonDictionary = (NSMutableDictionary *)response;
+        
+        MSIDAADV2TokenResponse *tokenResponse = (MSIDAADV2TokenResponse *)[self.oauth2Factory tokenResponseFromJSON:jsonDictionary context:nil error:&error];
```

Comment:
Try to use a local error for the whole block in this sendWithBlock, instead of re-using the "error" you got in line 170

Reply:
+1

Reply:
done